nbdev/clean.py

```diff
@@ -52,29 +53,36 @@ def _clean_cell_output(cell):
             o.get('metadata', {}).pop('tags', None)
 
 # %% ../nbs/11_clean.ipynb 8
-def _clean_cell(cell, clear_all=False):
+def _clean_cell(cell, clear_all=False, allowed_metadata_keys=None):
     "Clean `cell` by removing superfluous metadata or everything except the input if `clear_all`"
     if 'execution_count' in cell: cell['execution_count'] = None
     if 'outputs' in cell:
         if clear_all: cell['outputs'] = []
         else:         _clean_cell_output(cell)
     if cell['source'] == ['']: cell['source'] = []
     cell['metadata'] = {} if clear_all else {
-        k:v for k,v in cell['metadata'].items() if k=="hide_input"}
+        k:v for k,v in cell['metadata'].items() if k in allowed_metadata_keys}
 
 # %% ../nbs/11_clean.ipynb 9
-def clean_nb(nb, clear_all=False):
+def clean_nb(nb, clear_all=False,
+             allowed_metadata_keys:List[str]=None,
+             allowed_cell_metadata_keys:List[str]=None):
     "Clean `nb` from superfluous metadata"
-    for c in nb['cells']: _clean_cell(c, clear_all=clear_all)
-    nb['metadata'] = {k:v for k,v in nb['metadata'].items() if k in
-                     ("kernelspec", "jekyll", "jupytext", "doc")}
+    metadata_keys = {"kernelspec", "jekyll", "jupytext", "doc"}
+    if allowed_metadata_keys:
```

Comment:
I don't feel strongly about it, but we prefer to use a single line for 1-line conditionals and loops. Details here: https://docs.fast.ai/dev/style.html